modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -249,6 +249,9 @@ void  PngDecoder::readDataFromBuf( void* _png_ptr, unsigned char* dst, size_t si
 
 bool  PngDecoder::readHeader()
 {
+    if (m_frame_count > 1)
```

Comment:
i can add a note if needed
`if (m_frame_count > 1) // this means readHeader() called before and no need to be called again`

Reply:
It makes sense to add it as comment for the condition.

Reply:
is good ?
```
if (m_frame_count > 1 /* if true, it means readHeader() was called before */)
{
    return true;
}
```